File: samples/python/hosted-agents/code-interpreter-custom/cleanup.py
"""
Cleanup script for code-interpreter-custom sample.

This script deletes agent versions created during the demo to prevent
accumulation of unused resources.

Usage:
    python cleanup.py <agent_name> <agent_version>
    python cleanup.py --all  # Deletes all versions of MyAgent

Environment Variables:
    AZURE_AI_PROJECT_ENDPOINT: Your AI Project endpoint
    AGENT_NAME (optional): Default agent name to clean up
    AGENT_VERSION (optional): Specific version to clean up
"""

import os
import sys
import argparse

import dotenv
from azure.ai.projects import AIProjectClient
from azure.identity import DefaultAzureCredential
from azure.core.exceptions import ResourceNotFoundError

dotenv.load_dotenv()


def delete_agent_version(project_client: AIProjectClient, agent_name: str, agent_version: str) -> bool:
    """
    Delete a specific agent version.
    
    Args:
        project_client: The AI Project client
        agent_name: Name of the agent
        agent_version: Version to delete
        
    Returns:
        True if deleted successfully, False otherwise
    """
    try:
        print(f"🗑️  Deleting agent '{agent_name}' version '{agent_version}'...")
        project_client.agents.delete_version(agent_name=agent_name, agent_version=agent_version)
        print(f"✅ Successfully deleted agent '{agent_name}' version '{agent_version}'")
        return True
    except ResourceNotFoundError:
        print(f"⚠️  Agent '{agent_name}' version '{agent_version}' not found (may have been already deleted)")
        return False
    except Exception as e:
        print(f"❌ Error deleting agent '{agent_name}' version '{agent_version}': {e}")
        return False


def list_agent_versions(project_client: AIProjectClient, agent_name: str) -> list:
    """
    List all versions of an agent.
    
    Args:
        project_client: The AI Project client
        agent_name: Name of the agent
        
    Returns:
        List of agent versions
    """
    try:
        versions = list(project_client.agents.list_versions(agent_name=agent_name))
        return versions
    except ResourceNotFoundError:
        print(f"⚠️  Agent '{agent_name}' not found")
        return []
    except Exception as e:
        print(f"❌ Error listing versions for agent '{agent_name}': {e}")
        return []


def delete_all_versions(project_client: AIProjectClient, agent_name: str) -> int:
    """
    Delete all versions of an agent.
    
    Args:
        project_client: The AI Project client
        agent_name: Name of the agent
        
    Returns:
        Number of versions deleted
    """
    print(f"📋 Listing all versions of agent '{agent_name}'...")
    versions = list_agent_versions(project_client, agent_name)
    
    if not versions:
        print(f"ℹ️  No versions found for agent '{agent_name}'")
        return 0
    
    print(f"Found {len(versions)} version(s) to delete")
    deleted_count = 0
    
    for version in versions:
        if delete_agent_version(project_client, agent_name, version.version):
            deleted_count += 1
    
    return deleted_count


def main():
    parser = argparse.ArgumentParser(
        description="Clean up agent versions created by code-interpreter-custom sample"
    )
    parser.add_argument(
        "agent_name",
        nargs="?",
        default=os.getenv("AGENT_NAME", "MyAgent"),
        help="Name of the agent to clean up (default: MyAgent or AGENT_NAME env var)"
    )
    parser.add_argument(
        "agent_version",
        nargs="?",
        default=os.getenv("AGENT_VERSION"),
        help="Specific version to delete (optional, defaults to AGENT_VERSION env var)"
    )
    parser.add_argument(
        "--all",
        action="store_true",
        help="Delete all versions of the specified agent"
    )
    
    args = parser.parse_args()
    
    # Validate environment
    if not os.getenv("AZURE_AI_PROJECT_ENDPOINT"):
        print("❌ Error: AZURE_AI_PROJECT_ENDPOINT environment variable is required")
        sys.exit(1)
    
    # Create project client
    try:
        project_client = AIProjectClient(
            endpoint=os.environ["AZURE_AI_PROJECT_ENDPOINT"],
            credential=DefaultAzureCredential(),
        )
    except Exception as e:
        print(f"❌ Error creating project client: {e}")
        sys.exit(1)
    
    print("🧹 Code Interpreter Custom - Resource Cleanup")
    print("=" * 60)
    
    with project_client:
        if args.all:
            # Delete all versions
            deleted = delete_all_versions(project_client, args.agent_name)
            print("=" * 60)
            print(f"✅ Cleanup complete: {deleted} version(s) deleted")
        elif args.agent_version:
            # Delete specific version
            success = delete_agent_version(project_client, args.agent_name, args.agent_version)
            print("=" * 60)
            if success:
                print("✅ Cleanup complete")
            else:
                print("⚠️  Cleanup completed with warnings")
        else:
            print("❌ Error: Please specify an agent version or use --all flag")
            print("\nUsage examples:")
            print(f"  python cleanup.py {args.agent_name} <version>")
            print(f"  python cleanup.py {args.agent_name} --all")
            sys.exit(1)


if __name__ == "__main__":
    main()

File: samples/typescript/enterprise-agent-tutorial/1-idea-to-prototype/src/cleanup.ts
#!/usr/bin/env node
/**
 * Cleanup script for enterprise-agent-tutorial sample
 * 
 * This script deletes agents and conversations created during demos
 * to prevent accumulation of unused resources.
 * 
 * Usage:
 *   npm run cleanup
 *   node dist/cleanup.js <agent_name> <agent_version>
 *   node dist/cleanup.js --conversation <conversation_id>
 *   node dist/cleanup.js --all
 * 
 * Environment Variables:
 *   PROJECT_ENDPOINT: Your Azure AI Foundry project endpoint
 *   AGENT_NAME (optional): Default agent name to clean up
 *   AGENT_VERSION (optional): Default agent version to clean up
 *   CONVERSATION_ID (optional): Conversation ID to clean up
 */

import { AgentsClient } from "@azure/ai-agents-ii";
import { getBearerTokenProvider, DefaultAzureCredential } from "@azure/identity";
import OpenAI from "openai";
import { config } from "dotenv";

config();

interface CleanupOptions {
  agentName?: string;
  agentVersion?: string;
  conversationId?: string;
  deleteAll?: boolean;
}

const projectEndpoint = process.env.PROJECT_ENDPOINT || "";

async function deleteAgentVersion(
  agentsClient: AgentsClient,
  agentName: string,
  agentVersion: string
): Promise<boolean> {
  try {
    console.log(`🗑️  Deleting agent '${agentName}' version '${agentVersion}'...`);
    await agentsClient.deleteVersion(agentName, agentVersion);
    console.log(`✅ Successfully deleted agent '${agentName}' version '${agentVersion}'`);
    return true;
  } catch (error: any) {
    if (error.statusCode === 404) {
      console.log(`⚠️  Agent '${agentName}' version '${agentVersion}' not found (may have been already deleted)`);
    } else {
      console.error(`❌ Error deleting agent '${agentName}' version '${agentVersion}': ${error.message}`);
    }
    return false;
  }
}

async function deleteConversation(
  openAIClient: OpenAI,
  conversationId: string
): Promise<boolean> {
  try {
    console.log(`🗑️  Deleting conversation '${conversationId}'...`);
    await openAIClient.conversations.delete(conversationId);
    console.log(`✅ Successfully deleted conversation '${conversationId}'`);
    return true;
  } catch (error: any) {
    if (error.status === 404) {
      console.log(`⚠️  Conversation '${conversationId}' not found (may have been already deleted)`);
    } else {
      console.error(`❌ Error deleting conversation '${conversationId}': ${error.message}`);
    }
    return false;
  }
}

async function listAgentVersions(
  agentsClient: AgentsClient,
  agentName: string
): Promise<any[]> {
  try {
    const versions = [];
    for await (const version of agentsClient.listVersions(agentName)) {
      versions.push(version);
    }
    return versions;
  } catch (error: any) {
    if (error.statusCode === 404) {
      console.log(`⚠️  Agent '${agentName}' not found`);
    } else {
      console.error(`❌ Error listing versions for agent '${agentName}': ${error.message}`);
    }
    return [];
  }
}

async function deleteAllVersions(
  agentsClient: AgentsClient,
  agentName: string
): Promise<number> {
  console.log(`📋 Listing all versions of agent '${agentName}'...`);
  const versions = await listAgentVersions(agentsClient, agentName);

  if (versions.length === 0) {
    console.log(`ℹ️  No versions found for agent '${agentName}'`);
    return 0;
  }

  console.log(`Found ${versions.length} version(s) to delete`);
  let deletedCount = 0;

  for (const version of versions) {
    if (await deleteAgentVersion(agentsClient, agentName, version.version!)) {
      deletedCount++;
    }
  }

  return deletedCount;
}

async function cleanup(options: CleanupOptions): Promise<void> {
  console.log("🧹 Enterprise Agent Tutorial - Resource Cleanup");
  console.log("=".repeat(60));

  if (!projectEndpoint) {
    console.error("❌ Error: PROJECT_ENDPOINT environment variable is required");
    process.exit(1);
  }

  try {
    const credential = new DefaultAzureCredential();

    // Create Agents client
    const agentsClient = new AgentsClient(projectEndpoint, credential, {
      apiVersion: "2025-05-15-preview",
    });

    // Create OpenAI client for conversation cleanup
    const scope = "https://ai.azure.com/.default";
    const azureADTokenProvider = await getBearerTokenProvider(credential, scope);
    const openAIClient = new OpenAI({
      apiKey: azureADTokenProvider,
      baseURL: `${projectEndpoint}/openai`,
      defaultQuery: { "api-version": "2025-05-15-preview" },
    });

    let cleanupPerformed = false;

    // Delete conversation if specified
    if (options.conversationId) {
      await deleteConversation(openAIClient, options.conversationId);
      cleanupPerformed = true;
    }

    // Delete agent version(s)
    if (options.deleteAll && options.agentName) {
      const deleted = await deleteAllVersions(agentsClient, options.agentName);
      console.log("=".repeat(60));
      console.log(`✅ Cleanup complete: ${deleted} version(s) deleted`);
      cleanupPerformed = true;
    } else if (options.agentName && options.agentVersion) {
      const success = await deleteAgentVersion(
        agentsClient,
        options.agentName,
        options.agentVersion
      );
      console.log("=".repeat(60));
      if (success) {
        console.log("✅ Cleanup complete");
      } else {
        console.log("⚠️  Cleanup completed with warnings");
      }
      cleanupPerformed = true;
    }

    if (!cleanupPerformed) {
      console.log("❌ Error: No cleanup operations specified");
      console.log("\nUsage examples:");
      console.log("  npm run cleanup -- --agent workplace-assistant --version 1");
      console.log("  npm run cleanup -- --agent workplace-assistant --all");
      console.log("  npm run cleanup -- --conversation <conversation_id>");
      console.log("  npm run cleanup -- --agent workplace-assistant --version 1 --conversation <id>");
      process.exit(1);
    }
  } catch (error: any) {
    console.error(`\n❌ Cleanup failed: ${error.message}`);
    console.error("Please check your configuration and credentials");
    process.exit(1);
  }
}

// Parse command line arguments
function parseArgs(): CleanupOptions {
  const args = process.argv.slice(2);
  const options: CleanupOptions = {
    agentName: process.env.AGENT_NAME,
    agentVersion: process.env.AGENT_VERSION,
    conversationId: process.env.CONVERSATION_ID,
  };

  for (let i = 0; i < args.length; i++) {
    const arg = args[i];
    
    if (arg === "--agent" && i + 1 < args.length) {
      options.agentName = args[++i];
    } else if (arg === "--version" && i + 1 < args.length) {
      options.agentVersion = args[++i];
    } else if (arg === "--conversation" && i + 1 < args.length) {
      options.conversationId = args[++i];
    } else if (arg === "--all") {
      options.deleteAll = true;
    } else if (!arg.startsWith("--")) {
      // Positional arguments: agent_name agent_version
      if (!options.agentName) {
        options.agentName = arg;
      } else if (!options.agentVersion) {
        options.agentVersion = arg;
      }
    }
  }

  return options;
}

// Main execution
const options = parseArgs();
cleanup(options);
